internal/consensus/common_test.go

```diff
@@ -397,6 +399,32 @@ func subscribeToVoter(ctx context.Context, t *testing.T, cs *State, addr []byte)
 	return ch
 }
 
+func subscribeToVoterBuffered(ctx context.Context, t *testing.T, cs *State, addr []byte) <-chan tmpubsub.Message {
+	t.Helper()
+	votesSub, err := cs.eventBus.Subscribe(context.Background(), testSubscriber, types.EventQueryVote, 10)
+	if err != nil {
+		t.Fatalf("failed to subscribe %s to %v", testSubscriber, types.EventQueryVote)
+	}
+	ch := make(chan tmpubsub.Message)
+	go func() {
+		for {
+			msg, err := votesSub.Next(ctx)
+			if err != nil {
+				if !errors.Is(err, pubsub.ErrTerminated) && !errors.Is(err, context.Canceled) {
+					t.Fatalf("error retrieving value of subscription %s", err)
```

Comment:
```suggestion
					t.Errorf("error retrieving value of subscription %s", err)
```
This is in a different goroutine from the test, so it can't safely fatal (which terminates the goroutine, but not the right one)